### PR TITLE
add CLI to introspect state of daemon's `NameManager` field

### PR DESCRIPTION
**Pull request by ianvernon**
This will list the internal state of all DNS names to poll, as well as the mapping of FQDNSelector to regular expression:
    
```
    $ cilium fqdn names
    {
      "FQDNPolicySelectors": [
        {
          "selectorString": "MatchName: api.twitter.com, MatchPattern: ",
          "regexString": "^api[.]twitter[.]com[.]$"
        }
      ],
      "DNSPollNames": [
        "api.twitter.com."
      ]
    }
```

Signed-off by: Ian Vernon <ian@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9132)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

File: Documentation/foo.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+ID uint16
```
Comment (joestringer) [inline Documentation/foo.txt]: What is `foo.txt`?

**Review by joestringer — CHANGES_REQUESTED** [review]
nit

File: Documentation/foo.txt (hunk 1 above)
Comment (ianvernon) [reply Documentation/foo.txt]: I did `git add -u Documentation/` and my laziness got punished - fixed now. Sorry about that!

**Review by nebril — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Comment by ianvernon** [comment]
test-me-please

File: api/v1/openapi.yaml [hunk 2]
```diff
@@ -783,7 +783,23 @@ paths:
             "$ref": "#/definitions/Error"
         '404':
           description: No DNS data with provided parameters found
-
+  "/fqdn/names":
+    get:
+      summary: List internal DNS selector representations
+      description: |
+        Retrieves the list of DNS-related fields (names to poll, selectors and
+        their corresponding regexes.
```
Comment (aanm) [inline api/v1/openapi.yaml]: Missing right parentheses

**Review by aanm — COMMENTED** [review]
This is not blocking but `cilium fqdn names` is something that is exposed to users. Wouldn't be better if we pick user friendly names in the user side? I'm talking about the `allSelectors` and `namesToPoll`. Suggestion to `FQDNPolicySelectors` and `DNSPollNames`

File: api/v1/openapi.yaml (hunk 2 above)
Comment (ianvernon) [reply api/v1/openapi.yaml]: 👍 , having easy to understand names for end users is important. Will be fixed in next upload!

Comment (ianvernon) [reply api/v1/openapi.yaml]: Fixed!

**Comment by ianvernon** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by ianvernon** [comment]
Istio test failure: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/14969/

Appears unrelated. Trying again.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Pull request merged**